### PR TITLE
Allow to unconditionally hide status bar in app mode

### PR DESCRIPTION
**Pull request by dos1**
Make it configurable, as this makes the status bar inaccessible.

---

FWIW, I am using this together with gesture-based navigation to effectively hide both status and navigation bars while in app mode, keeping the back button available as a gesture.

### CODE REVIEW

**Comment by aleasto** [comment]
I had tried to filter-out the navigation bar in this same way, since they also take up invisible hitbox space in multi-windows mode.
But for some reason that caused maximized windows to stop responding to mouse or touch inputs.

Anyways, this is fine

**Comment by aleasto** [comment]
Maybe use `SystemProperties::getBoolean(String key, boolean default)`

File: waydroid-patches/base-patches-30/frameworks/base/0036-Force-hide-status-bar-in-multi-windows-mode.patch [hunk 1]
```diff
@@ -52,7 +52,7 @@ index 81378831adbc..4e68edc08aea 100644
 +    public static boolean shouldForceHide(@InternalInsetsType int type) {
 +        return (type == ITYPE_CAPTION_BAR ||
```
Comment (aleasto) [inline waydroid-patches/base-patches-30/frameworks/base/0036-Force-hide-status-bar-in-multi-windows-mode.patch]: hold on, this was never meant to be `ITYPE_CAPTION_BAR`, it was meant to be `ITYPE_CLIMATE_BAR`.

Interesting that this didn't totally break the caption bar...

**Comment by dos1** [comment]
> I had tried to filter-out the navigation bar in this same way

The first thing I tried as well :)

Updated to use `getBoolean`. Haven't tested this version though (got rid of my build tree for now).

**Pull request merged**